### PR TITLE
feat: add authentication

### PR DESCRIPTION
**Pull request by anik31**

### CODE REVIEW

**Review by Maianki — APPROVED** [review]
Great job.  Your code is clean and modular you can go ahead with the merge.

File: src/features/auth/authSlice.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    token: null,
+    user: null,
+};
+
+export const loginUser = createAsyncThunk("auth/login", async(credentials, { rejectWithValue })=>{
+  try{
+    const {data} = await axios.post("/api/auth/login", credentials);
+    return data;
+  }catch(error){
+    return rejectWithValue(error.response.data.errors[0]);
```
Comment (sainath7878) [inline src/features/auth/authSlice.js:17]: Appreciation: Good that you're showing the error message returned from API

**Review by sainath7878 — APPROVED** [review]
Code is clean and understandable. Awesome work on the UI.

File: src/features/auth/authSlice.js (hunk 1 above)
Comment (anik31) [reply src/features/auth/authSlice.js:17]: Thanks

**Pull request merged**